WordPress/Classes/ViewRelated/System/Floating Create Button/CreateButtonActionSheet.swift

```diff
@@ -1,68 +1,21 @@
+protocol ActionSheetItem {
+    var handler: () -> Void { get }
+    func makeButton() -> ActionSheetButton
+}
+
 /// The Action Sheet containing action buttons to create new content to be displayed from the Create Button.
 class CreateButtonActionSheet: ActionSheetViewController {
 
     enum Constants {
         static let title = NSLocalizedString("Create New", comment: "Create New header text")
-
-        enum Badge {
-            static let font = UIFont.preferredFont(forTextStyle: .caption1)
-            static let insets = UIEdgeInsets(top: 2, left: 8, bottom: 2, right: 8)
-            static let cornerRadius: CGFloat = 2
-            static let backgroundColor = UIColor.muriel(color: MurielColor(name: .red, shade: .shade50))
-        }
     }
 
-    init(newPost: @escaping () -> Void, newPage: @escaping () -> Void, newStory: (() -> Void)?) {
-        let postsButton = CreateButtonActionSheet.makePostsButton(handler: newPost)
-        let pagesButton = CreateButtonActionSheet.makePagesButton(handler: newPage)
-        let storiesButton = CreateButtonActionSheet.makeStoriesButton(handler: { newStory?() })
-        let shouldShowStories = newStory != nil
-        let buttons = shouldShowStories ? [postsButton, pagesButton, storiesButton] : [postsButton, pagesButton]
-
+    init(actions: [ActionSheetItem]) {
+        let buttons = actions.map({ return $0.makeButton() })
```

Comment:
Maybe `let buttons = actions.map { $0.makeButton() }`  just to make the syntax a little crisper?

Reply:
👍 Yep! I was hoping to use like a KeyPath here, but that doesn't seem to work yet.